spec/mailers/reminder_deadline_mailer_spec.rb

```diff
@@ -17,18 +23,30 @@
     end
 
     it 'renders the body' do
-      deadline_date = "#{Date.current.year}-#{Date.current.month}-#{organization.deadline_day}".to_date.strftime('%a, %d %b %Y')
+      deadline_date = today.change(day: organization.deadline_day).strftime('%a, %d %b %Y')
+
       expect(subject.body).to include("This is a friendly reminder that #{organization.name} requires your human essentials requests to be submitted by #{deadline_date}")
     end
 
+    context 'when the reminder day is greater than the deadline day' do
+      before { organization.update(deadline_day: today.prev_day.day) }
+
+      it 'specifies the deadline as being in the next month' do
+        deadline_date = today.next_month.change(day: organization.deadline_day).strftime('%a, %d %b %Y')
+
+        expect(subject.body).to include("This is a friendly reminder that #{organization.name} requires your human essentials requests to be submitted by #{deadline_date}")
+      end
+    end
+
     context "when the partner gets its deadline reminder data from a partner group association" do
       before do
-        partner_group = create(:partner_group, deadline_day_of_month: organization.deadline_day - 1)
+        partner_group = create(:partner_group, reminder_day_of_month: today.day, deadline_day_of_month: organization.deadline_day - 1)
         partner_group.partners << partner
       end
 
       it 'renders the body with the partner group deadline day' do
-        deadline_date = "#{Date.current.year}-#{Date.current.month}-#{partner.partner_group.deadline_day_of_month}".to_date.strftime('%a, %d %b %Y')
+        deadline_date = today.change(day: partner.partner_group.deadline_day_of_month).strftime('%a, %d %b %Y')
+
```

Comment:
If we do make a new service class, I feel like most of these specs should move to that. Really they're testing the calculation of the deadline date, not anything related to e-mail.

Reply:
Agreed.

Reply:
Done.